Source/WebCore/dom/EventTarget.h

```diff
@@ -50,16 +50,29 @@ namespace WebCore {
 
 struct AddEventListenerOptions;
 class DOMWrapperWorld;
+class EventTarget;
 class JSEventListener;
 
 struct EventTargetData {
     WTF_MAKE_NONCOPYABLE(EventTargetData); WTF_MAKE_FAST_ALLOCATED;
 public:
     EventTargetData() = default;
+
+    void clear()
+    {
+        eventListenerMap.clearEntriesForTearDown();
+    }
+
     EventListenerMap eventListenerMap;
 };
 
-class EventTarget : public ScriptWrappable, public CanMakeWeakPtr<EventTarget> {
+class EventTargetCounter : public EventTargetData {
```

Comment:
I'd call this EventTargetDataWithCounter.
But really, I don't think we even need to have this counter concept in most WeakPtrImpl.
We can have this defined only in a specialized version in the tests.

Reply:
Renamed to `WeakPtrImplWithEventTargetData`.